echopop/live/live_survey.py

```diff
@@ -0,0 +1,343 @@
+
+        # Get any acoustic files created
+        if "acoustic_files" in self.meta["provenance"]:
+            # ---- Get the filenames
+            acoustic_filenames = self.meta["provenance"]["acoustic_files_read"]
+            # ---- Subset if many files are being processed
+            if len(acoustic_filenames) > 2:
+                acoustic_filenames = (
+                    acoustic_filenames[:2] + ["..."] + [f"[n = {len(acoustic_filenames)}]"]
+                )
+            # ---- Format string
+            acoustic_files = ", ".join(acoustic_filenames)
+        else:
+            acoustic_files = "None"
+
+        # Get any biology files created
+        if "biology_files" in self.meta["provenance"]:
+            # ---- Get the filenames
+            biology_filenames = self.meta["provenance"]["biology_files_read"]
+            # ---- Subset if many files are being processed
+            if len(biology_filenames) > 4:
+                biology_filenames = biology_filenames + ["..."]
+            # ---- Format string
+            biology_files = ", ".join(biology_filenames)
+        else:
+            biology_files = "None"
+
+        # Get linked database names
+        linked_dbs = "\n   ".join(
+            [f"{key.title()}: {db}" for key, db in self.config["database"].items()]
+        )
+
+        return (
+            f"LiveSurvey-class object \n"
+            f"Timestamp: {self.meta['date']} \n"
+            f"Acoustic files being processed: \n   {acoustic_files}\n"
+            f"Biology files being processed: \n   {biology_files}\n"
+            f"Linked databases: \n   {linked_dbs}"
+        )
+
+    def __str__(self):
+        return self.__repr__()
+
+    def load_acoustic_data(
+        self, xarray_kwargs: dict = {}, input_filenames: Optional[list] = None, verbose: bool = True
+    ):
+
+        # Validate the data directory and format the filepaths
+        acoustic_files = eldl.validate_data_directory(
+            self.config, dataset="acoustics", input_filenames=input_filenames
+        )
+
+        # Read in the acoustic data files
+        if acoustic_files:
+            # ! [REQUIRES DASK] ---- Read in the listed file
+            # ---- Read in the acoustic data files
+            prc_nasc_df, acoustic_data_units = eldl.read_acoustic_files(
+                acoustic_files, xarray_kwargs=xarray_kwargs
+            )
+            # ---- Add the `acoustic_data_units` to the dictionary
+            self.config["acoustics"]["dataset_units"] = acoustic_data_units
+            # ---- Preprocess the acoustic dataset
+            # TODO: SettingWithCopyWarning:
+            self.input["acoustics"]["prc_nasc_df"] = preprocess_acoustic_data(
+                prc_nasc_df.copy(), self.input["spatial"], self.config
+            )
+            # ---- Add meta key
+            self.meta["provenance"].update(
+                {
+                    "acoustic_files_read": acoustic_files,
+                }
+            )
```

Comment:
These additions are ad-hoc and deviates from the idea behind a validated config model.